errors.lua

```diff
@@ -382,6 +382,85 @@ local function netbox_call(conn, func_name, ...)
     )
 end
 
+
+local function _wrap_future_result(prefix, suffix, res, err)
+    if err == nil then
+        return res
+    end
+
+    if not is_error_object(err) then
+        err = NetboxCallError:new(err)
+    end
+
+    return _wrap(prefix, suffix, nil, err)
+end
+
+--- Perform protected wait of async net.box call.
+-- Note: function passed to net.box call must not return multiple values.
+-- It should return only tuple res, error (also raising an error is acceptable)
+-- Similar to `netbox_call`,
+-- execute code on remote server using Tarantool built-in [`net.box` `conn:call` `is_async = true`](
+-- https://tarantool.io/en/doc/latest/reference/reference_lua/net_box/#net-box-call).
+-- Additionally postprocess returned values with `wrap`.
+-- @see netbox_call
+-- @function netbox_wait_async_call
+-- @param future net.box future object
+-- @tparam number timeout
+-- @tparam ?string call_uri - async call uri (to increase error verbosity)
+-- @tparam ?string call_func_name - async call function name (to increase error verbosity)
+-- @return[1] Postprocessed `future:wait_result()` result
+-- @treturn[2] nil
+-- @treturn[2] error_object Error description
+local function netbox_wait_async_call(future, timeout, call_uri, call_func_name)
+    if type(future) ~= 'table' then
+        error('Bad argument #1 to errors.netbox_wait_async_call' ..
+            ' (net.box future expected, got ' .. type(future) .. ')', 2)
+    elseif type(timeout) ~= 'number' then
+        error('Bad argument #2 to errors.netbox_wait_async_call' ..
+            ' (number expected, got ' .. type(timeout) .. ')', 2)
+    elseif call_uri and type(call_uri) ~= 'string' then
+        error('Bad argument #3 to errors.netbox_wait_async_call' ..
+            ' (?string expected, got ' .. type(call_uri) .. ')', 2)
+    elseif call_func_name and type(call_func_name) ~= 'string' then
+        error('Bad argument #4 to errors.netbox_wait_async_call' ..
+        ' (?string expected, got ' .. type(call_func_name) .. ')', 2)
+    end
+
+    if future.method == 'eval' then
+        error("errors.netbox_wait_async_call doesn't support eval", 2)
```

Comment:
Why? Are there any objective reasons?

Reply:
eval allows (without error):
```lua
conn:eval("return 1,2,3", nil)
```
and error handling will fail